src/Microsoft.DotNet.ProjectModel/Graph/LockFileReaderCache.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.DotNet.ProjectModel.Graph
+{
+    public class LockFileReaderCache
+    {
+        private readonly Dictionary<string, LockFile> _cache = new Dictionary<string, LockFile>();
+
+        public LockFile Read(string lockFilePath)
+        {
+            lock (_cache)
```

Comment:
Use a different thing to look it by `private readonly object lock _cacheLock = new object();`